### ISSUE TITLE
Severe performance degradation after upgrade from 7.24.0 to 7.26.0

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-che/che#19112]
### Describe the bug
I have upgraded my local Che installation from 7.24.0 to 7.26.0 yesterday and this morning, I started seeing the new dashboard (not yesterday after the update, though). 
When starting a self-hosting workspace today (see che-theia doc), I am experiencing 80% CPU load and I never get to a state where I can enter text into a terminal in Che theia.
What I've tried:

1. Remove all plugins from the workspace. No effect
2. Use a che-theia meta.yml that does not have CPU limits. No effect
3. Use che-theia latest or next: no effect
The only error message in the browser log that wasn't there yesterday is this one:
```
The connection to wss://serverpruq8ci0-jwtproxy-server-4400.10.20.30.61.nip.io/attach/1?token=eyJhbGciOiJSUzI1NiIsImtpbmQiOiJtYWNoaW5lX3Rva2VuIiwia2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyJ9.eyJ3c2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyIsInVpZCI6IjJmYjAxNmNlLTZlNjMtNGQ5Yi04MTE0LWExMTQ5MjY5OThjNiIsImF1ZCI6IndvcmtzcGFjZTY2ZzZwNmpkdW93OWdkanMiLCJuYmYiOi0xLCJ1bmFtZSI6ImFkbWluIiwiaXNzIjoid3NtYXN0ZXIiLCJleHAiOjE2NDUyNjc1MjEsImlhdCI6MTYxMzczMTUyMSwianRpIjoiZjYyMWQwY2MtNzE4NC00ZWVkLWIxOWUtNzRmNjc2YjdhZmE5In0.MZwLtmtJnef1tawLMBnxPw7BwP06m8nxnyEEgReQD5Xjf6U9VLv7sKjrKEO8u4rI8HGqXxPobIgdS8JXmCJKBw-gaReKye3_-hVI_Hc2S9pNE3spLe8pDoWE0wi8n2Nt7AU_-0Fe2d4Xgnrjl340O2qD3qBgi0PhReyfZrbCFg3N2ySA1Q3YTv4THgOOKMV-QIWpSTKlIs0HDvO9U-isRqkHrI9fGDRDQ6_PXGexubbCWFcMs8ea64pzhGpTWunlXbMgQAxWPF8nbGhqE9PDYWKNfN8nrthSrHg-UMwEa1hVxFkJb7_Rq696eP5nBxfcXcF0fkGz4CZ3XYx3-4y1bw was interrupted while the page was loading.
```

But that probably just indicates that the back-end is not answering. Inspection of the theia container logs didn't reveal anything out of the ordinary.

### Che version

 7.26.0
### Runtime
   - [ ] kubernetes (include output of `kubectl version`)
   - [ ] Openshift (include output of `oc version`)
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [x] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

### Installation method
   - [x] chectl server:update
     * provide a full command that was used to deploy Eclipse Che (including the output)
     * provide an output of `chectl version` command
   - [ ] OperatorHub
   - [ ] I don't know

### Environment
   - [ ] my computer
       - [x] Windows
       - [ ] Linux
       - [ ] macOS
   - [ ] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [ ] other (please specify)
   - [ ] other: please specify

**Comment by tsmaeder** [comment]
Note that the back-end is consuming up to 50% of my CPU. That is the maximum it's can use in my docker configuration.

**Comment by amisevsk** [comment]
@tsmaeder is it the Theia backend or the Dashboard/Che-server backend consuming the CPU? Does the issue occur if you load the workspace directly (i.e. without the dashboard frame)?

I'm tentatively marking this one as `area/che-theia`

**Comment by benoitf** [comment]
It doesn't look to be tied to che-theia (che.openshift.io using che 7.26.0)

> 'I am experiencing 80% CPU load'

CPU is consumed by the eclipse che workspace pod or by other services ?

what are the pods/containers that consume cpu ?

what is the output of `docker stats` command. ?

did you update docker desktop as well ?

**Comment by tsmaeder** [comment]
docker stats says: 

```
CONTAINER ID   NAME                                                                                                                               CPU %     MEM USAGE / LIMIT     MEM %     NET I/O   BLOCK I/O   PIDS
b0aa2ba92bee   k8s_theia-ide5mr_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1           137.49%   218.5MiB / 1GiB       21.34%    0B / 0B   0B / 0B     50
939862fea245   k8s_che-machine-execkb3_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1    0.71%     13.87MiB / 128MiB     10.84%    0B / 0B   0B / 0B     15
e47017f3ca10   k8s_che-dev_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1                0.00%     5.914MiB / 6GiB       0.10%     0B / 0B   0B / 0B     7
04c0074ab6be   k8s_plugin-host_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1            0.01%     1.887MiB / 1GiB       0.18%     0B / 0B   0B / 0B     1
86ccdce07f34   k8s_che-jwtproxyx9hl5uci_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1   5.28%     8.988MiB / 128MiB     7.02%     0B / 0B   0B / 0B     15
f602f9adaa1d   k8s_che_che-dc86cc5bd-t74bw_eclipse-che_6395eed3-4914-4384-9f12-8b06cc491c0b_7                                                     2.79%     364.9MiB / 1GiB       35.63%    0B / 0B   0B / 0B     104
8cee73533f9e   k8s_che-plugin-registry_plugin-registry-556d8f979c-wbb7b_eclipse-che_4d1eee40-bee1-4980-b9f0-47e7dbbe27b0_5                        0.00%     7.758MiB / 256MiB     3.03%     0B / 0B   0B / 0B     82
5d3fad06f92e   k8s_POD_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1                    0.00%     1.188MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
1a240e4b1c6c   k8s_keycloak_keycloak-85f49f97f6-nb656_eclipse-che_6290384e-2653-4e84-8582-63baa55e9ed1_5                                          0.07%     409MiB / 1.5GiB       26.63%    0B / 0B   0B / 0B     55
7d5f95cd08e5   k8s_che-devfile-registry_devfile-registry-7f76d9764f-wpc2v_eclipse-che_7b996799-9980-41d1-bcf5-50422b1dc467_5                      0.01%     8.93MiB / 256MiB      3.49%     0B / 0B   0B / 0B     82
ecc877e33aba   k8s_postgres_postgres-6697fcbd7f-shlj8_eclipse-che_be78a67a-bf07-45c5-aaa8-96854232d544_5                                          0.19%     45.3MiB / 1GiB        4.42%     0B / 0B   0B / 0B     12
6d39c961661c   k8s_POD_plugin-registry-556d8f979c-wbb7b_eclipse-che_4d1eee40-bee1-4980-b9f0-47e7dbbe27b0_5                                        0.00%     1.316MiB / 24.94GiB   0.01%     0B / 0B   0B / 0B     1
9a47991e47ff   k8s_POD_che-dc86cc5bd-t74bw_eclipse-che_6395eed3-4914-4384-9f12-8b06cc491c0b_5                                                     0.00%     1.18MiB / 24.94GiB    0.00%     0B / 0B   0B / 0B     1
ab027ab4f5a2   k8s_POD_devfile-registry-7f76d9764f-wpc2v_eclipse-che_7b996799-9980-41d1-bcf5-50422b1dc467_5                                       0.00%     1.012MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
3d8c4de26fc2   k8s_POD_keycloak-85f49f97f6-nb656_eclipse-che_6290384e-2653-4e84-8582-63baa55e9ed1_5                                               0.00%     1.176MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
a7b3b051c84d   k8s_POD_postgres-6697fcbd7f-shlj8_eclipse-che_be78a67a-bf07-45c5-aaa8-96854232d544_5                                               0.00%     1.023MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
47790a9162b7   k8s_che-operator_che-operator-557b9786fd-pxx56_eclipse-che_c7d28951-39b2-46ac-a699-3798bb19b49a_6                                  1.46%     16.61MiB / 256MiB     6.49%     0B / 0B   0B / 0B     21
29d1a276c6e6   k8s_nginx-ingress-controller_nginx-ingress-controller-9899c7dfc-wjvwq_ingress-nginx_de35ed9f-1ca7-4517-a15f-beac2b3569eb_21        6.84%     156.1MiB / 24.94GiB   0.61%     0B / 0B   0B / 0B     291
bae70b9789ee   k8s_coredns_coredns-f9fd979d6-t99sh_kube-system_f85d8e61-448d-429b-afa2-c1a903735c9a_21                                            0.39%     16.68MiB / 170MiB     9.81%     0B / 0B   0B / 0B     13
5ce2bfed2d42   k8s_vpnkit-controller_vpnkit-controller_kube-system_3cb7d487-76b5-43cd-a677-b1e106e0395c_21                                        0.00%     20.69MiB / 24.94GiB   0.08%     0B / 0B   0B / 0B     19
e03acb618337   k8s_storage-provisioner_storage-provisioner_kube-system_b2be0d6a-9e27-4718-a09b-126c2f25d310_42                                    0.05%     10.9MiB / 24.94GiB    0.04%     0B / 0B   0B / 0B     11
298c3356da59   k8s_coredns_coredns-f9fd979d6-pqjbz_kube-system_f10afe2f-7a73-4c0f-908b-3bc1b01ea1b4_21                                            0.39%     16.78MiB / 170MiB     9.87%     0B / 0B   0B / 0B     13
606d3950aace   k8s_kube-proxy_kube-proxy-cgl94_kube-system_f052c3dc-6fa9-435e-b169-e38cecdefe1c_21                                                0.04%     20.32MiB / 24.94GiB   0.08%     0B / 0B   0B / 0B     11
1d13865a93bb   k8s_POD_che-operator-557b9786fd-pxx56_eclipse-che_c7d28951-39b2-46ac-a699-3798bb19b49a_6                                           0.00%     908KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
cd0083090290   k8s_POD_nginx-ingress-controller-9899c7dfc-wjvwq_ingress-nginx_de35ed9f-1ca7-4517-a15f-beac2b3569eb_21                             0.00%     1.129MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
3117f91a0490   k8s_POD_storage-provisioner_kube-system_b2be0d6a-9e27-4718-a09b-126c2f25d310_21                                                    0.00%     1.066MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
a83b9dad2193   k8s_POD_vpnkit-controller_kube-system_3cb7d487-76b5-43cd-a677-b1e106e0395c_21                                                      0.00%     1.074MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
71a7b68041ac   k8s_POD_coredns-f9fd979d6-t99sh_kube-system_f85d8e61-448d-429b-afa2-c1a903735c9a_21                                                0.00%     1.023MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
0ae4ac26d91b   k8s_POD_coredns-f9fd979d6-pqjbz_kube-system_f10afe2f-7a73-4c0f-908b-3bc1b01ea1b4_21                                                0.00%     1.016MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
b44c5cbab788   k8s_POD_kube-proxy-cgl94_kube-system_f052c3dc-6fa9-435e-b169-e38cecdefe1c_21                                                       0.00%     852KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
e8ff904beeac   k8s_etcd_etcd-docker-desktop_kube-system_c7cc6a3c3118f127f5fd469ef69477e0_21                                                       4.11%     43.5MiB / 24.94GiB    0.17%     0B / 0B   0B / 0B     23
60111c3c73fd   k8s_kube-apiserver_kube-apiserver-docker-desktop_kube-system_2085f5be68201a12aeffb940f7ba4272_21                                   4.22%     285.1MiB / 24.94GiB   1.12%     0B / 0B   0B / 0B     14
a39c4d723214   k8s_kube-controller-manager_kube-controller-manager-docker-desktop_kube-system_07d42b7bf3823b30f57219558fcb55e2_21                 3.40%     51.37MiB / 24.94GiB   0.20%     0B / 0B   0B / 0B     12
af780750ef45   k8s_kube-scheduler_kube-scheduler-docker-desktop_kube-system_ee4c94eb845abf1878fb3c4c489b1365_33                                   0.84%     21.63MiB / 24.94GiB   0.08%     0B / 0B   0B / 0B     13
0ea91dbebc17   k8s_POD_etcd-docker-desktop_kube-system_c7cc6a3c3118f127f5fd469ef69477e0_21                                                        0.00%     1.016MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
87bd802c7bf5   k8s_POD_kube-scheduler-docker-desktop_kube-system_ee4c94eb845abf1878fb3c4c489b1365_21                                              0.00%     996KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
6a2eaa9732e6   k8s_POD_kube-controller-manager-docker-desktop_kube-system_07d42b7bf3823b30f57219558fcb55e2_21                                     0.00%     1.012MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
c08a2b058a9e   k8s_POD_kube-apiserver-docker-desktop_kube-system_2085f5be68201a12aeffb940f7ba4272_21                                              0.00%     1.027MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
```

**Comment by tsmaeder** [comment]
CPU is 50% docker VM (that's all it's allowed to use on my machine)

**Comment by benoitf** [comment]
and inside thea-ide container, what are the processes consuming CPU ? nodejs or other processes ?

you might execute command using `kubectl exec` if theia in not so responsive.

**Comment by tsmaeder** [comment]
And a bit later, docker stats shows


<details>
```
CONTAINER ID   NAME                                                                                                                               CPU %     MEM USAGE / LIMIT     MEM %     NET I/O   BLOCK I/O   PIDS
b0aa2ba92bee   k8s_theia-ide5mr_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1           6.95%     186.8MiB / 1GiB       18.25%    0B / 0B   0B / 0B     49
939862fea245   k8s_che-machine-execkb3_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1    0.05%     13.64MiB / 128MiB     10.66%    0B / 0B   0B / 0B     15
e47017f3ca10   k8s_che-dev_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1                0.02%     5.914MiB / 6GiB       0.10%     0B / 0B   0B / 0B     7
04c0074ab6be   k8s_plugin-host_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1            0.00%     1.887MiB / 1GiB       0.18%     0B / 0B   0B / 0B     1
86ccdce07f34   k8s_che-jwtproxyx9hl5uci_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1   0.00%     8.582MiB / 128MiB     6.70%     0B / 0B   0B / 0B     15
f602f9adaa1d   k8s_che_che-dc86cc5bd-t74bw_eclipse-che_6395eed3-4914-4384-9f12-8b06cc491c0b_7                                                     0.61%     378.4MiB / 1GiB       36.95%    0B / 0B   0B / 0B     109
8cee73533f9e   k8s_che-plugin-registry_plugin-registry-556d8f979c-wbb7b_eclipse-che_4d1eee40-bee1-4980-b9f0-47e7dbbe27b0_5                        0.01%     7.766MiB / 256MiB     3.03%     0B / 0B   0B / 0B     82
5d3fad06f92e   k8s_POD_workspace66g6p6jduow9gdjs.workspace-649db84b99-fsnhg_eclipse-che_bf6ec7ca-9861-476c-bfc6-26a843d13ea3_1                    0.00%     1.188MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
1a240e4b1c6c   k8s_keycloak_keycloak-85f49f97f6-nb656_eclipse-che_6290384e-2653-4e84-8582-63baa55e9ed1_5                                          0.17%     409.2MiB / 1.5GiB     26.64%    0B / 0B   0B / 0B     55
7d5f95cd08e5   k8s_che-devfile-registry_devfile-registry-7f76d9764f-wpc2v_eclipse-che_7b996799-9980-41d1-bcf5-50422b1dc467_5                      0.00%     8.953MiB / 256MiB     3.50%     0B / 0B   0B / 0B     82
ecc877e33aba   k8s_postgres_postgres-6697fcbd7f-shlj8_eclipse-che_be78a67a-bf07-45c5-aaa8-96854232d544_5                                          1.95%     45.44MiB / 1GiB       4.44%     0B / 0B   0B / 0B     12
6d39c961661c   k8s_POD_plugin-registry-556d8f979c-wbb7b_eclipse-che_4d1eee40-bee1-4980-b9f0-47e7dbbe27b0_5                                        0.00%     1.316MiB / 24.94GiB   0.01%     0B / 0B   0B / 0B     1
9a47991e47ff   k8s_POD_che-dc86cc5bd-t74bw_eclipse-che_6395eed3-4914-4384-9f12-8b06cc491c0b_5                                                     0.00%     1.18MiB / 24.94GiB    0.00%     0B / 0B   0B / 0B     1
ab027ab4f5a2   k8s_POD_devfile-registry-7f76d9764f-wpc2v_eclipse-che_7b996799-9980-41d1-bcf5-50422b1dc467_5                                       0.00%     1.012MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
3d8c4de26fc2   k8s_POD_keycloak-85f49f97f6-nb656_eclipse-che_6290384e-2653-4e84-8582-63baa55e9ed1_5                                               0.00%     1.176MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
a7b3b051c84d   k8s_POD_postgres-6697fcbd7f-shlj8_eclipse-che_be78a67a-bf07-45c5-aaa8-96854232d544_5                                               0.00%     1.023MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
47790a9162b7   k8s_che-operator_che-operator-557b9786fd-pxx56_eclipse-che_c7d28951-39b2-46ac-a699-3798bb19b49a_6                                  0.21%     16.75MiB / 256MiB     6.54%     0B / 0B   0B / 0B     21
29d1a276c6e6   k8s_nginx-ingress-controller_nginx-ingress-controller-9899c7dfc-wjvwq_ingress-nginx_de35ed9f-1ca7-4517-a15f-beac2b3569eb_21        1.06%     156.6MiB / 24.94GiB   0.61%     0B / 0B   0B / 0B     291
bae70b9789ee   k8s_coredns_coredns-f9fd979d6-t99sh_kube-system_f85d8e61-448d-429b-afa2-c1a903735c9a_21                                            0.89%     16.3MiB / 170MiB      9.59%     0B / 0B   0B / 0B     13
5ce2bfed2d42   k8s_vpnkit-controller_vpnkit-controller_kube-system_3cb7d487-76b5-43cd-a677-b1e106e0395c_21                                        0.00%     20.7MiB / 24.94GiB    0.08%     0B / 0B   0B / 0B     19
e03acb618337   k8s_storage-provisioner_storage-provisioner_kube-system_b2be0d6a-9e27-4718-a09b-126c2f25d310_42                                    0.31%     10.94MiB / 24.94GiB   0.04%     0B / 0B   0B / 0B     11
298c3356da59   k8s_coredns_coredns-f9fd979d6-pqjbz_kube-system_f10afe2f-7a73-4c0f-908b-3bc1b01ea1b4_21                                            0.48%     16.64MiB / 170MiB     9.79%     0B / 0B   0B / 0B     13
606d3950aace   k8s_kube-proxy_kube-proxy-cgl94_kube-system_f052c3dc-6fa9-435e-b169-e38cecdefe1c_21                                                0.00%     20.25MiB / 24.94GiB   0.08%     0B / 0B   0B / 0B     11
1d13865a93bb   k8s_POD_che-operator-557b9786fd-pxx56_eclipse-che_c7d28951-39b2-46ac-a699-3798bb19b49a_6                                           0.00%     908KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
cd0083090290   k8s_POD_nginx-ingress-controller-9899c7dfc-wjvwq_ingress-nginx_de35ed9f-1ca7-4517-a15f-beac2b3569eb_21                             0.00%     1.129MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
3117f91a0490   k8s_POD_storage-provisioner_kube-system_b2be0d6a-9e27-4718-a09b-126c2f25d310_21                                                    0.00%     1.066MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
a83b9dad2193   k8s_POD_vpnkit-controller_kube-system_3cb7d487-76b5-43cd-a677-b1e106e0395c_21                                                      0.00%     1.074MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
71a7b68041ac   k8s_POD_coredns-f9fd979d6-t99sh_kube-system_f85d8e61-448d-429b-afa2-c1a903735c9a_21                                                0.00%     1.023MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
0ae4ac26d91b   k8s_POD_coredns-f9fd979d6-pqjbz_kube-system_f10afe2f-7a73-4c0f-908b-3bc1b01ea1b4_21                                                0.00%     1.016MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
b44c5cbab788   k8s_POD_kube-proxy-cgl94_kube-system_f052c3dc-6fa9-435e-b169-e38cecdefe1c_21                                                       0.00%     852KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
e8ff904beeac   k8s_etcd_etcd-docker-desktop_kube-system_c7cc6a3c3118f127f5fd469ef69477e0_21                                                       8.63%     47.75MiB / 24.94GiB   0.19%     0B / 0B   0B / 0B     23
60111c3c73fd   k8s_kube-apiserver_kube-apiserver-docker-desktop_kube-system_2085f5be68201a12aeffb940f7ba4272_21                                   8.06%     276.5MiB / 24.94GiB   1.08%     0B / 0B   0B / 0B     14
a39c4d723214   k8s_kube-controller-manager_kube-controller-manager-docker-desktop_kube-system_07d42b7bf3823b30f57219558fcb55e2_21                 4.32%     51.42MiB / 24.94GiB   0.20%     0B / 0B   0B / 0B     12
af780750ef45   k8s_kube-scheduler_kube-scheduler-docker-desktop_kube-system_ee4c94eb845abf1878fb3c4c489b1365_33                                   1.52%     21.11MiB / 24.94GiB   0.08%     0B / 0B   0B / 0B     13
0ea91dbebc17   k8s_POD_etcd-docker-desktop_kube-system_c7cc6a3c3118f127f5fd469ef69477e0_21                                                        0.00%     1.016MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
87bd802c7bf5   k8s_POD_kube-scheduler-docker-desktop_kube-system_ee4c94eb845abf1878fb3c4c489b1365_21                                              0.00%     996KiB / 24.94GiB     0.00%     0B / 0B   0B / 0B     1
6a2eaa9732e6   k8s_POD_kube-controller-manager-docker-desktop_kube-system_07d42b7bf3823b30f57219558fcb55e2_21                                     0.00%     1.012MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
c08a2b058a9e   k8s_POD_kube-apiserver-docker-desktop_kube-system_2085f5be68201a12aeffb940f7ba4272_21                                              0.00%     1.027MiB / 24.94GiB   0.00%     0B / 0B   0B / 0B     1
```
</details>

**Comment by tsmaeder** [comment]
I had manually killed the etcd container in docker desktop and that seems to have considerably improved things. I don't get the 50% usage from the back-end anymore, but I still can't open new terminal windows: I always get a pair of 

```
The connection to wss://server1lknga5n-jwtproxy-server-4400.10.20.30.61.nip.io/attach/2?token=eyJhbGciOiJSUzI1NiIsImtpbmQiOiJtYWNoaW5lX3Rva2VuIiwia2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyJ9.eyJ3c2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyIsInVpZCI6IjJmYjAxNmNlLTZlNjMtNGQ5Yi04MTE0LWExMTQ5MjY5OThjNiIsImF1ZCI6IndvcmtzcGFjZTY2ZzZwNmpkdW93OWdkanMiLCJuYmYiOi0xLCJ1bmFtZSI6ImFkbWluIiwiaXNzIjoid3NtYXN0ZXIiLCJleHAiOjE2NDUyNzk2NDUsImlhdCI6MTYxMzc0MzY0NSwianRpIjoiNjY3YjdlMTYtNzVjOS00NWM5LWFmZDctODkwZThmNDNlN2ZhIn0.fvXlryN4nFV1R8wdrthdjLMTV6zcunPEVJdDmRDMWMClKKx0EC5avphiZjdRrUDRzFzEAQ4M09YT5O-xaygwWbeQJvaabOFqeDTpoO6usT-zWEaxAr50ze7HRAlDeN_sDBrb-CcOTG6HxDKLQ5Mg3Z436yJPL0_OFqEgBZss4poLbVLRBU4FRaOdZq5-Q1WVALgkaH8sGLnZrvOdfkpDnvh8cC_CB4zu3h9S8d_qVN_nKl6PLMgUke8aF8N0RGx0oCmwcCJ5K_mfAKmPECXhDPlkYkBUq3SG6MwcbzFu29s6MDEPHe_eBqxkGdlOaSKsfU6UxGPD_ZEzR_0kWWxm7Q was interrupted while the page was loading. reconnecting-websocket-mjs.js:518:18
Firefox can’t establish a connection to the server at wss://server1lknga5n-jwtproxy-server-4400.10.20.30.61.nip.io/attach/2?token=eyJhbGciOiJSUzI1NiIsImtpbmQiOiJtYWNoaW5lX3Rva2VuIiwia2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyJ9.eyJ3c2lkIjoid29ya3NwYWNlNjZnNnA2amR1b3c5Z2RqcyIsInVpZCI6IjJmYjAxNmNlLTZlNjMtNGQ5Yi04MTE0LWExMTQ5MjY5OThjNiIsImF1ZCI6IndvcmtzcGFjZTY2ZzZwNmpkdW93OWdkanMiLCJuYmYiOi0xLCJ1bmFtZSI6ImFkbWluIiwiaXNzIjoid3NtYXN0ZXIiLCJleHAiOjE2NDUyNzk2NDUsImlhdCI6MTYxMzc0MzY0NSwianRpIjoiNjY3YjdlMTYtNzVjOS00NWM5LWFmZDctODkwZThmNDNlN2ZhIn0.fvXlryN4nFV1R8wdrthdjLMTV6zcunPEVJdDmRDMWMClKKx0EC5avphiZjdRrUDRzFzEAQ4M09YT5O-xaygwWbeQJvaabOFqeDTpoO6usT-zWEaxAr50ze7HRAlDeN_sDBrb-CcOTG6HxDKLQ5Mg3Z436yJPL0_OFqEgBZss4poLbVLRBU4FRaOdZq5-Q1WVALgkaH8sGLnZrvOdfkpDnvh8cC_CB4zu3h9S8d_qVN_nKl6PLMgUke8aF8N0RGx0oCmwcCJ5K_mfAKmPECXhDPlkYkBUq3SG6MwcbzFu29s6MDEPHe_eBqxkGdlOaSKsfU6UxGPD_ZEzR_0kWWxm7Q. reconnecting-websocket-mjs.js:518:18
```

**Comment by tsmaeder** [comment]
@benoitf "top" in the theia IDE container just shows the usual suspects:

```
   32    21 root     S     292m   1%   7   0% /usr/local/bin/node /home/theia/node_modules/@theia/core/lib/node/messaging/ipc-bootstrap.js --nsfwOptions={}
30075    21 root     S     282m   1%   7   0% /usr/local/bin/node /home/theia/node_modules/@eclipse-che/theia-remote/lib/node/plugin-host-custom.js
   21     1 root     S     331m   1%   6   0% node src-gen/backend/main.js /projects --hostname=127.0.0.1 --port=3100
30172 30075 root     S     221m   1%   2   0% /usr/local/bin/node /default-theia-plugins/vscode-builtin-json-language-features/extension/server/dist/node/js
```

**Comment by tsmaeder** [comment]
> Does the issue occur if you load the workspace directly (i.e. without the dashboard frame)?

@amisevsk how do I do that? If I just cut out the "dashboard" part of the URI, I get redirected to the dashboard.

**Comment by benoitf** [comment]
>> Does the issue occur if you load the workspace directly (i.e. without the dashboard frame)?

>@amisevsk how do I do that? If I just cut out the "dashboard" part of the URI, I get redirected to the dashboard.


you grab and connect to the `theia`URL endpoint of theia-ide component (from che workspace REST API if you have problem seeing theia and its container panel), or looking at the embeded iframe source location of the theia page from dashboard

**Comment by tsmaeder** [comment]
I've manually added cpuLimit = cpuRequest = 2000m and I was able to open a terminal again.

**Comment by tsmaeder** [comment]
So, to recap:

1. There was a problem with persistent, very high CPU. Killing etcd manually seems to have resolved that. Restarting docker-desktop and che multiple times did not have this effect. Maybe this was an artifact of the upgrade, I guess we'll never find out.
2. I have observed that adding the cpuLimits by hand seems to improve performance. Not sure how the defaults get established on docker-desktop, but I get the impression they worked better in 7.24.0 than in 7.26.0. 🤷 

Since I have a working environment again, I will reduce priority to p2.

**Comment by benoitf** [comment]
in 7.26.1 default memory has been changed to 1500m while it was 1000m in 7.26.0 https://github.com/eclipse/che-plugin-registry/commit/958f280b834999ac608d98fd2586d21c96274443#diff-4526f5cc968bddb01b7aaa7b95bbf3be602a7cfe9b7a69230ecbef43751732b4

**Comment by benoitf** [comment]
you might give a try to 7.26.1 :)

**Comment by tsmaeder** [comment]
@benoitf on a serious note: I used to run Che nightly in order to test our newest and greatest. But my workflow was broken so often that I've stopped doing that. That doing a simple update leads to such disruption distresses me: today _I_ am having such severe problems, but tomorrow it might be an important customer evaluating Che/CRW.
If we don't care about local installs like docker-desktop, we should stop pretending to support them. Not sure what that would do to our own development process, though.
